aea/connections/stub/connection.py

```diff
@@ -275,14 +275,6 @@ def from_config(
         output_file = configuration.config.get(
             OUTPUT_FILE_KEY, DEFAULT_OUTPUT_FILE_NAME
         )  # type: str
-        restricted_to_protocols_names = {
-            p.name for p in configuration.restricted_to_protocols
-        }
-        excluded_protocols_names = {p.name for p in configuration.excluded_protocols}
         return StubConnection(
-            input_file,
-            output_file,
-            connection_id=configuration.public_id,
-            restricted_to_protocols=restricted_to_protocols_names,
-            excluded_protocols=excluded_protocols_names,
+            input_file, output_file, address=address, configuration=configuration,
```

Comment:
Note, this was wrong, the configuration is needed to recover for instance the public. id